packages/core/src/behaviors.ts

```diff
@@ -462,7 +474,7 @@ export function createMachineBehavior<
     get initialState() {
       resolvedMachine =
         resolvedMachine || (isFunction(machine) ? machine() : machine);
-      return resolvedMachine.initialState; // TODO: this should get from machine.getInitialState(ref)
```

Comment:
it depends on how this is called by other things but it might be also important for this to be cached - need to recheck that

Reply:
i've just added caching for now plus a TODO comment to recheck this later